See: https://github.com/sagemath/cypari2/pull/165
and https://github.com/sagemath/sage/pull/38749

From 0a5a7b42e53d065f8d78bdaa8181d6afa20d1f4f Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Tue, 1 Oct 2024 18:54:19 +0200
Subject: [PATCH] Add pari_PRIMES declaration

Needed to port sagemath to work with pari 2.17
---
 cypari2/paridecl.pxd | 1 +
 cypari2/types.pxd    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/cypari2/paridecl.pxd b/cypari2/paridecl.pxd
index 29cb8ef..9ccf336 100644
--- a/cypari2/paridecl.pxd
+++ b/cypari2/paridecl.pxd
@@ -118,6 +118,7 @@ cdef extern from *:     # PARI headers already included by types.pxd
     extern PariOUT* pariOut
     extern PariOUT* pariErr
     extern byteptr diffptr
+    extern pari_prime* pari_PRIMES
 
     ###############################################
     #                                             #
diff --git a/cypari2/types.pxd b/cypari2/types.pxd
index 9ee4fe6..7f00b52 100644
--- a/cypari2/types.pxd
+++ b/cypari2/types.pxd
@@ -24,6 +24,7 @@ cdef extern from "pari/pari.h":
     ctypedef long* GEN
     ctypedef char* byteptr
     ctypedef unsigned long pari_sp
+    ctypedef unsigned long pari_prime
 
     # PARI types
     enum:
